tests/kernel/fatal/exception/src/main.c

```diff
@@ -71,7 +71,15 @@ void k_sys_fatal_error_handler(unsigned int reason, const z_arch_esf_t *pEsf)
 
 void entry_cpu_exception(void *p1, void *p2, void *p3)
 {
+#if defined(CONFIG_CPU_AARCH32_CORTEX_R) || defined(CONFIG_CPU_AARCH32_CORTEX_A)
+	expected_reason = K_ERR_ARM_UNDEFINED_INSTRUCTION;
+#elif defined(CONFIG_ARMV8_M_MAINLINE) && defined(CONFIG_USERSPACE)
+	expected_reason = K_ERR_ARM_USAGE_ILLEGAL_EPSR;
```

Comment:
I am having trouble determining under which conditions `K_ERR_ARM_USAGE_ILLEGAL_EPSR` is generated vs `K_ERR_ARM_MEM_INSTRUCTION_ACCESS`. My hunch is something to do with the MPU (maybe `z_arm_configure_dynamic_mpu_regions`?) Would appreciate if anyone could chime in.